### PR TITLE
PUD-1577: Add tests for authenticated clients

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/WebClientConfig.kt [hunk 1]
```diff
@@ -34,37 +34,27 @@ class WebClientConfig(
...
-  @Value("\${bankHolidayRegister.endpoint.url}")
-  private lateinit var bankHolidayRegisterEndpointUrl: String
-
   @Bean("bankHolidayRegisterWebClient")
-  fun bankHolidayRegisterWebClient(): TimeoutHandlingWebClient = timeoutHandlingWebClient(bankHolidayRegisterEndpointUrl)
-
-  @Value("\${courtRegister.endpoint.url}")
-  private lateinit var courtRegisterEndpointUrl: String
+  fun bankHolidayRegisterWebClient(@Value("\${bankHolidayRegister.endpoint.url}") bankHolidayRegisterEndpointUrl: String):
+    TimeoutHandlingWebClient = timeoutHandlingWebClient(bankHolidayRegisterEndpointUrl)
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/WebClientConfig.kt:39]: Cosmetic, but maybe drop the explicit return type (`: TimeoutHandlingWebClient`): the other equivalents below don't have it.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/WebClientConfig.kt [hunk 2]
```diff
@@ -34,37 +34,27 @@ class WebClientConfig(
       )
       .build()
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/WebClientConfig.kt:35]: Given we have `AuthenticatedWebClientConfig` maybe rename this class to e.g. `NoAuthWebClientConfig`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/search/PrisonerOffenderSearchClient.kt [hunk 3]
```diff
@@ -24,15 +23,13 @@ class PrisonerOffenderSearchClient(
 ) {
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/search/PrisonerOffenderSearchClient.kt:23]: Remove ^^^ `FIXME: PUD-1577  ...` comment above - assuming it is fixed by this PR.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/mockservers/PrisonApiMockServer.kt [hunk 4]
```diff
@@ -25,19 +25,24 @@ class PrisonApiMockServer(
...
-          aResponse()
-            .withHeaders(HttpHeaders(HttpHeader(CONTENT_TYPE, APPLICATION_JSON_VALUE)))
-            .withStatus(OK.value())
-            .withBody(objectMapper.writeValueAsString(responseBody))
-        )
-    )
+  fun latestMovementsRespondsWith(
+    request: Set<NomsNumber>,
+    responseBody: List<Movement>?,
+    withAuthorization: Boolean = true
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/mockservers/PrisonApiMockServer.kt:31]: How come we need a switch on auth here?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/mockservers/PrisonerOffenderSearchMockServer.kt [hunk 5]
```diff
@@ -38,18 +38,24 @@ class PrisonerOffenderSearchMockServer(
...
-          aResponse()
-            .withHeaders(HttpHeaders(HttpHeader(CONTENT_TYPE, APPLICATION_JSON_VALUE)))
-            .withStatus(OK.value())
-            .withBody(objectMapper.writeValueAsString(responseBody))
-        )
-    )
+  fun getPrisonerByNomsNumberRespondsWith(
+    nomsNumber: NomsNumber,
+    responseBody: Prisoner,
+    withAuthorization: Boolean = true
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/mockservers/PrisonerOffenderSearchMockServer.kt:44]: Ditto: How come we need a switch on auth here?

**Review by AndyNewton — COMMENTED** [review]
Few QQs.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/mockservers/PrisonApiMockServer.kt (hunk 4 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/mockservers/PrisonApiMockServer.kt]: As discussed this was needed as the Integration tests weren't setting the authorisation header.

**Review by AndyNewton — APPROVED** [review]

**Pull request merged**